src/libcore/ptr.rs

```diff
@@ -2808,14 +2808,14 @@ impl<T: ?Sized> fmt::Pointer for Unique<T> {
 #[unstable(feature = "ptr_internals", issue = "0")]
 impl<'a, T: ?Sized> From<&'a mut T> for Unique<T> {
     fn from(reference: &'a mut T) -> Self {
-        Unique { pointer: NonZero(reference as _), _marker: PhantomData }
+        Unique { pointer: NonZero(reference as *mut T as *const T), _marker: PhantomData }
```

Comment:
If I understand you correctly, the `_` here was filled in with `&'a T`, and that's no good.

Reply:
Yes.

Ideally we could make it infer `*mut T` for the `_`, that would be the safer choice.